File: csharp/ToolGood.Algorithm2/MathNet/Distributions/LogNormal.cs
using System;

namespace ToolGood.Algorithm.MathNet.Numerics.Distributions
{
    /// <summary>
    /// Continuous Univariate Log-Normal distribution.
    /// For details about this distribution, see
    /// <a href="http://en.wikipedia.org/wiki/Log-normal_distribution">Wikipedia - Log-Normal distribution</a>.
    /// </summary>
    class LogNormal
    {
        public static double CDF(double mu, double sigma, double x)
        {
            //if (sigma < 0.0) {
            //    throw new ArgumentException("InvalidDistributionParameters");
            //}

            return x < 0.0 ? 0.0
                : 0.5 * (1.0 + SpecialFunctions.Erf((Math.Log(x) - mu) / (sigma * Constants.Sqrt2)));
        }

        public static double InvCDF(double mu, double sigma, double p)
        {
            //if (sigma < 0.0) {
            //    throw new ArgumentException("InvalidDistributionParameters");
            //}

            return p <= 0.0 ? 0.0 : p >= 1.0 ? double.PositiveInfinity
                : Math.Exp(mu - sigma * Constants.Sqrt2 * SpecialFunctions.ErfcInv(2.0 * p));
        }
    }
}

File: JavaScript/src/MathNet/Distributions/LogNormal.js
import { Constants } from '../Constants.js';
import { SpecialFunctions } from '../SpecialFunctions/SpecialFunctions.js';

class LogNormal {
    /**
     * Log-Normal cumulative distribution function
     * @param {number} mu
     * @param {number} sigma
     * @param {number} x
     * @returns {number}
     */
    static cdf(mu, sigma, x) {
        return x < 0 ? 0
            : 0.5 * (1 + SpecialFunctions.erf((Math.log(x) - mu) / (sigma * Constants.sqrt2)));
    }

    /**
     * Log-Normal inverse cumulative distribution function
     * @param {number} mu
     * @param {number} sigma
     * @param {number} p
     * @returns {number}
     */
    static invCdf(mu, sigma, p) {
        return p <= 0 ? 0 : p >= 1 ? Infinity
            : Math.exp(mu - sigma * Constants.sqrt2 * SpecialFunctions.erfcInv(2 * p));
    }
}

export { LogNormal };